strawberryfields/apps/similarity.py

```diff
@@ -412,119 +527,218 @@ def prob_event_mc(
-    >>> feature_vector_sampling(samples, [2, 4, 6])
-    [0.19035, 0.2047, 0.1539]
+    >>> graph = nx.complete_graph(8)
+    >>> feature_vector_orbits(graph, [ [1,1], [2], [2,1,1], [1,1,1,1] ])
+    [0.18423136914879445,
+     7.843627302119571e-31,
+     0.03744399092424445,
+     0.07020748298295838]
+
+    >>> feature_vector_orbits(graph, [ [1,1], [2] , [2,1,1], [1,1,1,1] ], mc_samples = 1000)
+    [0.1842313691487924,
+     8.098317264668192e-31,
+     0.037443990924243886,
+     0.07020748298295804]
 
     Args:
-        samples (list[list[int]]): a list of samples
-        event_photon_numbers (list[int]): a list of events described by their total photon number
-        max_count_per_mode (int): maximum number of photons per mode for all events
+        graph (nx.Graph): input graph
+        list_of_orbits (list[list[int]]): a list of orbits
+        n_mean (float): total mean photon number of the GBS device
+        mc_samples (int): number of samples used in the Monte Carlo estimation
+        loss (float): fraction of photons lost in GBS
 
     Returns:
-        list[float]: a feature vector of event probabilities in the same order as
-        ``event_photon_numbers``
+        list[float]: a feature vector of orbit probabilities
     """
-    if min(event_photon_numbers) < 0:
-        raise ValueError("Cannot request events with photon number below zero")
 
-    if max_count_per_mode < 0:
-        raise ValueError("Maximum number of photons per mode must be non-negative")
-
-    n_samples = len(samples)
-
-    e = (sample_to_event(s, max_count_per_mode) for s in samples)
-    count = Counter(e)
+    if len(list_of_orbits) <= 0:
+        raise ValueError("List of orbits must have at least one orbit")
+    if all(isinstance(elem, list) for elem in list_of_orbits) == False:
+        raise ValueError("List of orbits should only consist of orbits")
+    if any(min(item) < 0 for item in list_of_orbits):
+        raise ValueError("Cannot request orbits with photon number below zero")
+    if n_mean < 0:
+        raise ValueError("Mean photon number must be non-negative")
+    if not 0 <= loss <= 1:
+        raise ValueError("Loss parameter must take a value between zero and one")
+    if mc_samples:
+        return [_prob_orbit_mc(graph, orbit, n_mean, mc_samples, loss) for orbit in list_of_orbits]
 
-    return [count[p] / n_samples for p in event_photon_numbers]
+    return [_prob_orbit_exact(graph, orbit, n_mean, loss) for orbit in list_of_orbits]
 
 
-def feature_vector_mc(
+def feature_vector_events(
     graph: nx.Graph,
     event_photon_numbers: list,
     max_count_per_mode: int = 2,
     n_mean: float = 5,
-    samples: int = 1000,
+    mc_samples: int = None,
     loss: float = 0.0,
 ) -> list:
-    r"""Calculates feature vector using Monte Carlo estimation of event probabilities according to the
-    input graph.
+    r"""Calculates feature vector made up of either exact or MC-approximated event probabilities for the input graph.

```

Comment:
I'd actually not mention exact vs MC in the one liner intro sentence, it can be mentioned in the main paragraph and this kept short and sweet.

Reply:
(similar to a previous comment, where @ixfoduap suggested not to abbreviate MC)

Reply:
see my comment on JM's comment